lib/ansible/modules/cloud/amazon/kinesis_firehose.py

```diff
@@ -0,0 +1,626 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: kinesis_firehose
+short_description: Manage a Kinesis Firehose Delivery Stream.
+description:
+    - Create or Delete a Kinesis Firehose Delivery Stream.
+version_added: "2.5"
+author: Stephen Clark (clarkst)
+options:
+  delivery_stream_name:
+    description:
+      - "The name of the Kinesis Firhose Delivery Stream you are managing."
+    default: None
+    required: true
+  state:
+    description:
+      - "Create or Delete the Kinesis Stream."
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  wait:
+    description:
+      - Wait for operation to complete before returning.
+    required: false
+    default: true
+  wait_timeout:
+    description:
+      - How many seconds to wait for an operation to complete before timing out.
+    required: false
+    default: 300
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+  # Basic creation example:
+  - name: Set up Kinesis Firehose Delivery Stream with a Kinesis Stream input and S3 destination , wait for the stream to become ACTIVE
+    kinesis_firehose:
+      delivery_stream_name: clarky-test-stream
+      delivery_stream_type: KinesisStreamAsSource
+      kinesis_stream_source_configuration:
+        kinesis_stream_a_r_n: arn:aws:kinesis:eu-west-1:XXXXXXXXXXXX:stream/playground-sclark-eu-west-1-Stream
```

Comment:
I know you're doing this because you need to convert to KinesisStreamARN, but these should be just handled as special cases, perhaps after converting to camel case.